### ISSUE TITLE
unified source code, config format & docker images

### ISSUE THREAD

**Issue by lianhao** [opea-project/GenAIComps#287]
Currently, for each microservice in GenAIComp, we see variant versions of source code and hence different docker images. 

Take dataprep as example. Now we have the following 7 variant versions of source code:

- using milvus as DB backend
- using pgvector as DB backend
- using pinecore as DB backend
- using qdrant as DB backend
- using langchain on redis DB
- using llama_index on redis DB
- using langchain_ray on redis DB

Each variant version of the above source code has its own full source code, its own unique config format, its own docker image, and hence it could results own K8S deployment helmchart/manifests. 

This could be a maintaining nightmare in the future. Suppose today we want to upgrade the RESTful API of dataprep microservice to add a new parameter, we need to update all the 7 variant versions of source code, and hence release 7 new docker images. It's really difficult to make API the same if we need to change 7 places just for adding a new parameter. 

We suggest refactor the 7 variant versions of code into a single unified source code variant, with a plugin mechanism to have different DB backend to write its own plugin. The dataprep source code can maintain one copy of RESTful API definitions and calls into different DB backend plugin, and hence only need one docker image. It's easy in python to discover and load plugin at runtime, and many project already has its own implementation, e.g. many OpenStack proejct uses [stevedore](https://pypi.org/project/stevedore/).

Also, different DB backend of dataprep now uses different config format, e.g. for dataprep-milvus we need to configure `MILVUS`, `MILVUS_PORT`, `COLLECTION_NAME`, for datapre-pgvector we need to configure `PG_CONNECTION_STRING`, `INDEX_NAME`, `CHUNK_SIZE`, `CHUNK_OVERLAP`. This means we must have different K8S deployment helmchart/manifests for different DB backend which also create duplicated effort.

We suggest to use a unified URL format of configuration for different DB backends, e.g `DB_BACKEND=milvus://<ip>:<port>?collection_name=name1`, `DB_BACKEND=pgvector://<ip>:<port>?index_name=abc&chunk_size=1000&chunk_overlap=10`. 

Other microservices have the similar issues.

In summary, we want to enhance the current each microservice implementation by:

- Using plugin mechanism, each backend could be implemented as a plugin, hence only need to release 1 docker image for each microservice.
- Unify the config format, so each microservice only needs a set of K8S helmchart/manifest for deployment.

**Comment by ftian1** [comment]
@lkk12014402  pls ensure the latest design meet this valid requirement.

**Comment by lkk12014402** [comment]
> [@lkk12014402](https://github.com/lkk12014402) pls ensure the latest design meet this valid requirement.

assign sihan to have a design to enhance

**Comment by xiguiw** [comment]
@lianhao 

I believe 1.2 code refactor resolved this issue. How do you think about this?

Could this issue be closed?
